### PR TITLE
[seal] Add new port

### PR DESCRIPTION
**Pull request by musaprg**
Add Microsoft SEAL as a new port of vcpkg.
https://github.com/microsoft/SEAL

I confirmed that this modification works well.
However, I mainly use macOS and Linux, so I don't know if it works as well on Windows in this PR.

### CODE REVIEW

**Comment by LilyWangL** [comment]
Hi @musaprg, thanks for your PR! If this port doesn't support Windows, please so these steps as below:
1. Add [vcpkg_fail_port_install](https://github.com/microsoft/vcpkg/blob/f5c732b40d43f062278f247036b773477823813b/docs/maintainers/vcpkg_fail_port_install.md#vcpkg_fail_port_install) to portfile.cmake.
2. Add the following code to _VCPKG_PATH/scripts/ci.baseline.txt_:
```
seal:x86-windows=fail
seal:x64-windows=fail
seal:x64-windows-static=fail
seal:x64-uwp=fail
seal:arm64-windows=fail
seal:arm-uwp=fail
```

**Comment by LilyWangL** [comment]
Thanks for your updates! The failures on Windows are not related to this changes. It should not block to merge this PR.

File: ports/seal/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    SHA512 198f75371c7b0b88066495a40c687c32725a033fd1b3e3dadde3165da8546d44e9eaa9355366dd5527058ae2171175f757f69189cf7f5255f51eba14c6f38b78
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}/native/src
+    PREFER_NINJA
+    OPTIONS
+        -DSEAL_LIB_BUILD_TYPE=${SEAL_LIB_BUILD_TYPE}
+)
```
Comment (vicroms) [inline ports/seal/portfile.cmake]: You can use CMake to build the Windows libraries by passing the `ALLOW_COMMAND_LINE_BUILD` option during configure.

```cmake
vcpkg_configure_cmake(
    SOURCE_PATH ${SOURCE_PATH}/native/src
    PREFER_NINJA
    OPTIONS
        -DALLOW_COMMAND_LINE_BUILD=ON
        -DSEAL_LIB_BUILD_TYPE=${SEAL_LIB_BUILD_TYPE}
)
```

**Review by vicroms — CHANGES_REQUESTED** [review]

File: ports/seal/portfile.cmake (hunk 1 above)
Comment (musaprg) [reply ports/seal/portfile.cmake]: Thanks. I fixed.

**Comment by vicroms** [comment]
Please remove this line 

```
vcpkg_fail_port_install(MESSAGE "${PORT} currently only supports Linux and Mac platform" ON_TARGET "Windows")
```

so CI can attempt to build the library on Windows

**Review by vicroms — APPROVED** [review]
LGTM

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**